### ISSUE TITLE
Revert strict type for positional variables (xs:integer → xs:positiveInteger)

### ISSUE THREAD

**Issue by ChristianGruen** [qt4cg/qtspecs#1162]
I feel that the decision to change `xs:integer` to `xs:positiveInteger` was a bit hasty (https://github.com/qt4cg/qtspecs/pull/1131#issuecomment-2051379262):

To be consistent, numerous other expressions and functions would need to be rewritten as well to use stricter types (arbitrary examples: the `count` clause; `$err:line-number` in the catch clause; the result type of `fn:string-to-codepoints`; positions in `array:get`, `fn:parse-integer`, etc.; the position parameter in HOF functions, and so on and on). We haven’t done so yet, and I seriously wonder what exactly we would win from the stricter types. In many cases, it would be reasonable to also define a strict upper limit, which is not possible with our types anyway.

Implementations may build heavily on the fact that `xs:integer` has been the default type for integer values in previous versions of the languages. For example, we use cached instances for the most small integer values, or we rewrite constructs with `xs:integer` to other constructs accepting the same type.

For all these challenges, as always, technical solutions exist, but the question is if there aren’t more interesting things to focus on than on such corner cases. Queries like `count(1 to 1000000000000)` are not supported by all implementations either although they may appear trivial to the ordinary user (by coincidence, it’s supported by BaseX, but I doubt it has been used a lot).

In short, I would like us to revert the change in https://github.com/qt4cg/qtspecs/pull/1131/commits/bba6e4f1067e0ef0779688622a58320a5298d440 and stick with `xs:integer`. If people feel bad about it, I would suggest discussing strict types in a much broader and general way.

**Comment by michaelhkay** [comment]
Fair enough. What happened was I was asked to specify the type for the new addition to XPath, I thought xs:positiveInteger was the obvious choice, and then I discovered there were other places that specified xs:integer (and other places, like the count clause, that say nothing). From an implementation perspective, `xs:int` is probably a natural choice, and of course any subtype of the spec-defined type is acceptable. But I agree that if we do it one place we should do it more systematically, e.g. return type of position() and last().

**Comment by michaelhkay** [comment]
Closed by #1177
